### PR TITLE
Create helper function to start deprecation of skillsetMask

### PR DESCRIPTION
**Pull request by Willdotwhite**
Speculative PR draft to remove use of bitwise logic; the scale and performance of the app means we don't really get a whole lot of value from it, but it complicated queries in a way that will be _very_ difficult to port to Mongo.

### CODE REVIEW

File: api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.gmtkgamejam.models
+
+import kotlin.math.pow
+
+enum class Skills(val bitwiseId: Int) {
+    `2DArt`(1),
```
Comment (ppmg) [inline api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt:6]: Small point: usually enum values are named in all caps.

File: api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    companion object {
+        private val map = values().associateBy(Skills::bitwiseId)
+        private fun fromInt(type: Int) = map[type]
+
+        fun fromBitwiseId(id: Int): List<Skills> {
+            val skills: ArrayList<Skills> = arrayListOf();
+
+            // Range should be 0..{number of Skills - 1}
+            for (i in 0..7) {
```
Comment (ppmg) [inline api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt:23]: Why not loop over values() and extract the bitwiseId value from each item in the loop?

Comment (Willdotwhite) [reply api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt:23]: That is a much cleaner solution, good call - I'll redo this loop to use that, cheers :)

File: api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt (hunk 1 above)
Comment (Willdotwhite) [reply api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt:6]: Ah, thanks for the spot - can't believe IDEA didn't tell me off for that!

File: api/src/main/kotlin/com/gmtkgamejam/models/Team.kt [hunk 3]
```diff
@@ -9,7 +9,9 @@ data class Team (
     var author: String,
     var authorId: String,
     var description: String,
+
     var skillsetMask: Int,
+    var skills: List<Skills>,
```
Comment (Willdotwhite) [inline api/src/main/kotlin/com/gmtkgamejam/models/Team.kt:14]: Adding this as a parameter means it'll be automatically serialised, which we could potentially use for the deprecation of the bitwise ID.

If not, it's still handy for debugging.

**Review by ppmg — DISMISSED** [review]

File: api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.gmtkgamejam.models
+
+enum class Skills(val bitwiseId: Int) {
+    `2D_ART`(1),
+    `3D_ART`(2),
```
Comment (awildbrysen) [inline api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt:5]: The ` around these seem a bit out of place. couldn't we just go for ART_2D and ART_3D?
It seems accessing these also has to happen with the backticks around it, which IntelliJ annoyingly does not place for you.
If I'm the only one that finds this annoying feel free to ignore this comment

**Review by awildbrysen — CHANGES_REQUESTED** [review]

File: api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt (hunk 4 above)
Comment (Willdotwhite) [reply api/src/main/kotlin/com/gmtkgamejam/models/Skills.kt]: Amended, thanks 👍

**Review by awildbrysen — DISMISSED** [review]

**Review by awildbrysen — APPROVED** [review]

**Pull request merged**